test/fixtures/rules/std_rule_cases/AM08.yml

```diff
@@ -24,6 +24,22 @@ test_fail_missing_multiple_joins_not_last:
   fail_str: "SELECT foo.a, bar.b FROM foo left join bar left join baz on foo.x = bar.y;\n"
   fix_str: "SELECT foo.a, bar.b FROM foo cross join bar left join baz on foo.x = bar.y;\n"
 
+test_fail_cross_join_with_where_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a;\n"
+  # Implementation of fixes is not yet done.
+  # fix_str: "SELECT foo.a, bar.b FROM foo JOIN bar OR foo.a = bar.b;\n"
+
+test_fail_cross_join_with_complex_where_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a AND foo.x = 3;\n"
+  # Implementation of fixes is not yet done.
+  # fix_str: "SELECT foo.a, bar.b FROM foo JOIN bar OR foo.a = bar.b WHERE foo.x = 3;\n"
+
+test_pass_cross_join_with_complex_where_clause_not_simplifiable:
+  pass_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a OR foo.x = 3;\n"
```

Comment:
In cases like this (where it's not a simple `AND` where clause, but the join is unqualified) - it's definitely simpler right now to just let it slide. I agree with not correcting it to a `CROSS JOIN`, but perhaps we make this rule a bit less useful by letting these slide... 🤔 

This specific case is probably more accurately a `LEFT JOIN` with some filtering - but detecting that could be quite convoluted. What's your thinking?

Reply:
I don't believe that I'll be able to build exhaustive test case list for autofixer. Avoiding crashes / unintended behavior changes for autofixed code should be the ultimate priority IMO.

Check is still valuable as it is.

I hope this is not a blocker for PR being merged.

Reply:
Leaving it as is seems fine to me. We could always improve the rule later to handle trickier cases.